### PR TITLE
Add blogpost for Galaxy JupyterLab

### PR DESCRIPTION
**Pull request by anuprulez**
ping @bgruening Thanks!

### CODE REVIEW

**Comment by bgruening** [comment]
@anuprulez we are now on the Galaxy Hub ... very similar procedure https://github.com/bgruening/galaxy-hub/tree/master/content/news

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
...totyping and developing end-to-end AI projects. Using such an infrastructure, long-running AI model training programs can be executed remotely. Trained models, represented in a standard [open neural network exchange](https://github.com/onnx/onnx) (ONNX) format and other resulting datasets are created in a Galaxy history. Other features include GPU support for faster training, support for machine learning packages such as [TensorFlow](https://www.tensorflow.org/) and [Scikit-learn](https://scikit-learn.org/stable/), Git integration for version control, the option of creating and executing pipelines of notebooks, the availability of multiple dashboards for monitoring compute resources and visualizations using [Bokeh](https://docs.bokeh.org/en/latest/), [Seaborn](https://seaborn.pydata.org/), [Matplotlib](https://matplotlib.org/), [Bqplots](https://github.com/bqplot/bqplot), [Voila](https://github.com/voila-dashboards/voila). In addition, the JupyterLab tool can also be used as a regular Galaxy tool in a [workflow](https://usegalaxy.eu/u/kumara/w/gpujupytool-imported-from-uploaded-file). These features make the JupyterLab notebook highly suitable for creating and managing AI projects.
```
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:16]: Don't make it Galaxy Europe specific, you can just say "public Galaxy servers with its compute infrastructures "

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
...access to GPU-enabled environments for developing AI models. [Google Colab](https://colab.research.google.com/) and [Kaggle Kernels](https://www.kaggle.com/) provide limited sessions which are as long as only 12 hours. In addition to shorter session time, the amount of memory provided by their sessions is variable and dependent on how much in the past a user has used them. In contrast, Galaxy's JupyterLab provides constant compute resources for unlimited time consisting of 20 GB RAM, 15 GB additional from one GPU, 7 vCPUs and 1 TB of storage for each session of JupyterLab providing a robust environment for developing AI models. In addition, it is possible to create long-running AI model training jobs from any JupyterLab notebook that get executed remotely on Galaxy's cluster. The resulting datasets become available a newly created Galaxy history and the JupyterLab session that started the training job can be safely exited. This feature of training AI models remotely is not available in Google Colab and Kaggle Kernels. Paid subscriptions of Google Colab, pro and pro+, provide a better set of compute resources but they come at a cost of EUR 9.25 and EUR 42.25 per month, respectively.
```
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:24]: its probably not unlimited, isn't it?

Comment (bgruening) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:24]: 1 TB? where does this come from?

Comment (bgruening) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:24]: > The resulting datasets become available a newly created Galaxy history

here is something missing

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
...nstalls the above-mentioned packages. The Docker container is downloaded by a [Galaxy interactive tool](https://github.com/usegalaxy-eu/galaxy/blob/release_22.05_europe/tools/interactive/interactivetool_ml_jupyter_notebook.xml) to make it available on Galaxy Europe. Having a Docker container running in the backend provides many security benefits as it interacts minimally with the remote computer's operating system. In addition, a non-root user inside the container provides additional security benefits. These benefits are important as users can execute arbitrary code on JupyterLab notebooks. The Docker container can separately be downloaded to any laptop or personal computer (having at least 20 GBs of space) or any other compute infrastructure from [Docker hub](https://hub.docker.com/layers/anupkumar/docker-ml-jupyterlab/galaxy-integration-0.2/images/sha256-e2d7e28a2f975523db0f5ac29c2e2ce3c7a35b061072098ad388d5b42ee86fba?context=repo) and used. If NVIDIA GPUs are available, the Docker container will automatically recognise them. Otherwise, it will run on CPUs. The scripts to run the container is mentioned in the [GitHub repository](https://github.com/ anuprulez/ml-jupyter-notebook).
```
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: https://github.com/usegalaxy-eu/galaxy/blob/release_22.05_europe/tools/interactive/interactivetool_ml_jupyter_notebook.xml -> upstream URL?

Comment (bgruening) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: move this repo to the EU repo? https://github.com/ anuprulez/ml-jupyter-notebook

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+
+### Use-cases
+A recent scientific publication that predicts infected regions of [COVID-19 CT scan images](https://www.sciencedirect.com/science/article/pii/S2666990021000069) is reproduced using multiple features of JupyterLab. In addition, [ColabFold](https://github.com/sokrypton/ColabFold), a faster implementation of [AlphaFold2](https://www.nature.com/articles/s41586-021-03819-2), can also be accessed in this notebook to predict the 3D structure of protein sequences. JupyterLab notebook is accessible in two ways - first as an interactive Galaxy tool and second by running the underlying docker container. In both ways, long-running training can be executed on Galaxy’s compute infrastructure.
+
```
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:34]: maybe add some nice images?

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md (hunk 3 above)
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: mention that a complete conda environment is setup and a user has access to the entire conda universe

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+### How to apply for this resource
+
+[Google form](http://usegalaxy.eu/gpu-request) should be used to apply for this resource. As this resource is intended for research purposes, an official University Email ID should be given. Once the request is approved by a Galaxy admin, the resource can be readily used on Galaxy Europe. Steps for application:
```
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:38]: I would not mind if the industry is using it ... we just want to forbid abuse. So official university or company emails ? or something like that?

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+1. Create an account on [Galaxy Europe](https://usegalaxy.eu/) using your official university
+email id
+2. Apply for accessing GPU Jupyterlab using this [Google form](http://usegalaxy.eu/gpu-request)
+3. Use your official university email id in the Google form. This resource is available only for
+research purposes.
+4. Wait up to 1-2 days to get the request approved.
```
Comment (bgruening) [inline _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:45]: remove the days please

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md (hunk 2 above)
Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:24]: > not sure how many days it is? 4 days? We can write several days if not clear

> its probably not unlimited, isn't it?

not sure how many days, 4 it is? we can write several days

Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:24]: > 1 TB? where does this come from?

disk_space

It shows 1TB at `/home/<<docker user>>` . Not sure if it is per session or shared by all users?

Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:24]: > Not sure if we need this entire paragraph here, we do not need to compete with them I think.

yes, right

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md (hunk 3 above)
Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: > https://github.com/usegalaxy-eu/galaxy/blob/release_22.05_europe/tools/interactive/interactivetool_ml_jupyter_notebook.xml -> upstream URL?

Once the PR is merged then yes

Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: > move this repo to the EU repo? https://github.com/ anuprulez/ml-jupyter-notebook

Yes, sure

Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: The package/plugin that lets user to install conda env does not work well with JupyterLab. It was too buggy, we tried this one: https://pypi.org/project/jupyter-conda/ 

But now, there is also a different one which we can try: https://github.com/mamba-org/gator. Not sure how buggy that would be

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md (hunk 6 above)
Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:45]: for a few days

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md (hunk 5 above)
Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:38]: Ok, updated

**Comment by anuprulez** [comment]
closing this in favour of https://github.com/galaxyproject/galaxy-hub/pull/1832

File: _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md (hunk 3 above)
Comment (bgruening) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: its not about the plugin, you can still type "!conda install ....". Correct?

Comment (anuprulez) [reply _posts/2023-01-24-gpu-jupyterlab-in-galaxy.md:29]: Yes, that's possible :)

**Comment by anuprulez** [comment]
All the comments here have been fixed in https://github.com/galaxyproject/galaxy-hub/pull/1832. So, let's use the blogpost markdown for development at Galaxy-hub repo.

**Comment by anuprulez** [comment]
Closing this one as https://github.com/galaxyproject/galaxy-hub/pull/1832 has been merged

**Pull request closed without merging**